### PR TITLE
switch from using Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by bender2k14**
Approving this PR should be all you need to do start using GitHub actions.

I started with the YAML file that GitHub suggested when I created a GitHub Action through the repository website, then I made these changes.
- Changed the `pull_request` setting to match what Mark Karpov did in [GitHub actions for Haskell CI](https://markkarpov.com/post/github-actions-for-haskell-ci.html)
- Changed the value of `runs-on` to `windows-latest` in preparation for PR #210
- Changed the version of the .NET Core SDK to 2.1.4 (to match `.travis.yml`)
- Added a step to execute `./validate.sh` (to match `.travis.yml`)
- Added `--configuration Release` to the `build` and `test` steps (to match `.travis.yml`)
- Added `--no-build` to the `test` step (to match `.travis.yml`)

I tested what happens when
- a build fails because of a failing test (I got an email, but I had to read the logs to see which tests failed),
- when a PR is created (a build is started for the potential merge commit), and
- when the branch in a PR is updated (a build is started for the new potential merge commit).

In particular, I tested this after merging in the branch from PR #210.

I didn't change the name at the top of the YAML file or the name of the YAML file.  Maybe someone cares what those names are.

### CODE REVIEW

**Comment by bender2k14** [comment]
> Shouldn't we also add a .NET Framework build to it? So we can see that it really solves the problem we have with Travis CI?

I tested this in my fork of this repository.  That is what I was trying to share when I said this.

> I tested what happens when
> 
> * a build fail because of a failing test (I got an email, but I had to read the logs to see which tests failed),
> * when a PR is created (a build is started for the potential merge commit), and
> * when the branch in a PR is updated (a build is started for the new potential merge commit).
> 
> In particular, I tested this after merging in the branch for PR #210.

I now realize that I could have made this clearer by linking to [the GitHub Action that I created in my fork](https://github.com/bender2k14/fsharp-hedgehog/actions).  In particular, [this execution of the Action](https://github.com/bender2k14/fsharp-hedgehog/actions/runs/172930167) tests that this Action will work in this repository even after completing PR #210.

I think this PR can be completed, then you should see a successful build for PR #210.  Then that PR can be completed as well :)

**Comment by moodmosaic** [comment]
Then, I believe, once we merge this, we'd want to also update the status badge icon on the [README](https://github.com/hedgehogqa/fsharp-hedgehog/blob/master/README.md), to show the status of GitHub Actions build instead of Travis CI.

**Comment by moodmosaic** [comment]
Or basically, we can do this right here since `travis.yml` is removed. I can push to your branch if you want, after I figure out how to show the status of a GitHub Actions build.

**Comment by bender2k14** [comment]
Oh, sure.  I will add it quick to this branch.  Just a second.

**Comment by bender2k14** [comment]
I added the change to this branch in a force pushed commit.  The badge [looks like this on my fork](https://github.com/bender2k14/fsharp-hedgehog#fsharp-hedgehog--).

2020-07-20_08-04-07_183_chrome

I will say again...
> I didn't change the name at the top of the YAML file or the name of the YAML file. Maybe someone cares what those names are.

...that I didn't change the name of the Action, which is now displayed in this badge.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-fsharp-hedgehog [![NuGet][nuget-shield]][nuget] [![Travis][travis-shield]][travis]
+fsharp-hedgehog [![NuGet][nuget-shield]][nuget] ![](https://github.com/bender2k14/fsharp-hedgehog/workflows/.NET%20Core/badge.svg)
```
Comment (moodmosaic) [inline README.md:1]:
```suggestion
fsharp-hedgehog [![NuGet][nuget-shield]][nuget] ![](https://github.com/hedgehogqa/fsharp-hedgehog/workflows/.NET%20Core/badge.svg)
```

Comment (moodmosaic) [reply README.md:1]: I guess we want this to point at @hedgehogqa right after we merge this.

**Comment by moodmosaic** [comment]
:tada:

https://github.com/hedgehogqa/fsharp-hedgehog/actions

**Pull request merged**